webapp/channels/src/packages/mattermost-redux/src/actions/integrations.ts

```diff
@@ -37,16 +38,33 @@ export function getIncomingHook(hookId: string) {
     });
 }
 
-export function getIncomingHooks(teamId = '', page = 0, perPage: number = General.PAGE_SIZE_DEFAULT) {
-    return bindClientFunc({
-        clientFunc: Client4.getIncomingWebhooks,
-        onSuccess: [IntegrationTypes.RECEIVED_INCOMING_HOOKS],
-        params: [
-            teamId,
-            page,
-            perPage,
-        ],
-    });
+export function getIncomingHooks(teamId = '', page = 0, perPage: number = General.PAGE_SIZE_DEFAULT, includeTotalCount = false): ActionFuncAsync<IncomingWebhook[] | IncomingWebhooksWithCount> {
```

Comment:
You might be able to define this like the `searchTeams` action (https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/packages/mattermost-redux/src/actions/teams.ts#L171) to make it so that TS can figure out what the correct return type is here. You could probably define it as
```ts
export function getIncomingHooks(teamId = '', page = 0, perPage = DEFAULT_PAGE_SIZE, includeTotalCount?: false): ActionFuncAsync<IncomingWebhook[]>;
export function getIncomingHooks(teamId = '', page = 0, perPage = DEFAULT_PAGE_SIZE, includeTotalCount: true): ActionFuncAsync<IncomingWebhooksWithCount>;
export function getIncomingHooks(teamId = '', page = 0, perPage = DEFAULT_PAGE_SIZE, includeTotalCount = false): ActionFuncAsync {
    ....
}
```

If you have to fight with TypeScript too much to make it work though, don't worry about it. I remember it being hard to type `searchTeams` correctly

Reply:
Good idea, I would really prefer to implement it similar to the suggested function. However, I could not get it to work because defining it as three functions does not allow the use of default parameters. This would make it more inconvenient for the caller of the function.

Reply:
I think it might be possible with some weird combination of typing things like `teamId: string | undefined` on the first few lines that just define the overloaded types, but that's totally fine. I know from personal experience how easy it is to burn a ton of time just to improve type checking slightly when it already works well enough 😅